### DISCUSSION TITLE
open_basedir restriction in effect, but searching in a directory I haven't asked it to.

### DISCUSSION THREAD

**Discussion by roja45** [mpdf/mpdf#1727] [Q&A]
Hello,

I get the following error:

_Error
Error detected. PDF file generation aborted: file_exists(): open_basedir restriction in effect. File(/usr/local/apache2/htdocs/tcpdf/images/) is not within the allowed path(s)_

I understand what the error means, but in this particular server that directory cannot be added.

My question is why is it looking in this directory.  This directory is not referenced anywhere in my code, or the mpdf code, so why it is trying to search there, and how do I stop mpdf from doing so.

I have found various things regarding K_PATH_IMAGES but that's not referenced anywhere in the mpdf code either.

Really quite stumped on this one.

Thanks

**Comment by finwe** [comment]
Are you sure you are using mPDF? your path says tcpdf and I don't recall anything about K_PATH_IMAGES being used anytime in history of mPDF. Maybe share more info about your setup and your code.

**Comment by roja45** [comment]
Yes, I am sure I am using mPDF, but I can't be sure what other things are installed on the server.

For reference, this is a Prestashop installation, and my module uses mPDF, but it's possible there are other modules installed using tcpdf.

The question I have however is why it finding and using this path, even if it is being defined elsewhere?

Everything in my module is self-contained.  I import it and use it.  I have never had any issues previously, but in this installation (customer installation, so I have no control over it), I am getting this error, and I don't know how to prevent mpdf from looking in this directory.

```php
require_once _PS_MODULE_DIR_ . "module/libraries/mpdf/vendor/autoload.php";

$mpdf_config = [
    'mode' => '+aCJK',
    "autoScriptToLang" => true,
    "autoLangToFont" => true,
    "tempDir" => $temp_dir,
];

$this->mpdf = new \Mpdf\Mpdf($mpdf_config);
$this->mpdf->debug = false;
$this->mpdf->useSubstitutions = false;
$this->mpdf->simpleTables = false;
$this->mpdf->shrink_tables_to_fit = false;
```

**Reply by finwe** [reply]
You can find more information about used paths via Logging: https://mpdf.github.io/installation-setup/logging.html

**Reply by finwe** [reply]
Code examples without the actual HTML code are not really helpful. What is PS_MODULE_DIR constant contents? What is your mPDF version and what is the current open_basedir setting?

**Reply by roja45** [reply]
mPDF version is 8.0.2

PS_MODULE_DIR just has the base directory path, but I am just calling the mPDF /mpdf/vendor/autoload.php file.

open_basedir allowed paths: (/home/todoalm2:/home2/todoalm2:/usr/local/wepanel/filemanager:/usr/lib/php:/usr/local/lib/php:/tmp:/usr/local/bin/wp:/usr/local/bin/composer:/opt/alt/php73)

but what I am really trying to understand is where is this path coming from, /usr/local/apache2/htdocs/tcpdf/images/ and how is mPDF finding it.

what html code do you need to see?

**Reply by finwe** [reply]
Looking into the source code, the "Error detected" message comes from an Error Exception, which should contain the line and file of the original open_basedir error. If you see the whole backtrace of the exception, it will point you to the place where the incriminated file_exists is called at. We can continue from there.

**Reply by finwe** [reply]
What is the content of the `$temp_dir` variable?

**Reply by finwe** [reply]
I need to see any HTML code, or better to know if this error happens with a particular HTML used in mPDF, or even before the HTML is used, eg. during the constructor call. 

Anyway, something still seems off with your code example, because the Error detected exception is being thrown only when the debug setting is set to true, which is not by default and not changed in the example.
